plugins/modules/openssh_cert.py

```diff
@@ -344,6 +353,21 @@ def generate(self, module):
 
             try:
                 temp_directory = tempfile.mkdtemp()
+
+                # Private key is decrypted and stored in a temporary file when ``passphrase`` value is given
+                if self.passphrase:
+                    keypair = OpensshKeypair.load(
+                        self.signing_key,
+                        passphrase=self.passphrase.encode('UTF-8'),
+                        no_public_key=True
+                    )
+                    keypair.update_passphrase('')
+                    temp_pkey = tempfile.mkstemp(dir=temp_directory)[1]
+                    with open(temp_pkey, 'wb') as f:
```

Comment:
Simply writing an unprotected version of the key to disk is somewhat unsatisfactory. Is it possible to pass the password in by stdin instead?

Reply:
So no, OpenSSH added support to load public keys via stdin for fingerprint verificaiton with `-lf '-'`, but otherwise `ssh-keygen` just eats stdin. Any attempts I've made to pass in the data via run_command with/without `prompt_regex` just results in the `Enter passphrase:` prompt halting playbook execution. (I haven't seen OpenSSH acknowledge this behavior anywhere, but that seems to be the consensus from any info I've found)  I don't mind closing this if writing the unencrypted key is non-compliant.

Reply:
Hmm, I guess it's ok if there's a big fat warning in the option's description that this happens. It's really annoying that OpenSSH doesn't allow to properly script this.

Reply:
I think I'll leave this in draft for a little bit for the following reasons:
- This feature just facilitates automation, but it's nothing that user's wouldn't be able to do themselves (for community.crypto >= 1.7.0)
- In situations where you are permitted to just store CA/signing keys on disk I'm not sure encrypting the keys is a major concern. (i.e. if encryption is really required then you are probably beyond just using `ssh-keygen` with locally stored secrets)
- A more ideal solution would be certificate generation through `cryptography` for which all the mechanisms already exist (See [here](https://github.com/pyca/cryptography/blob/1b922ed1dee0cd7e165a868639ce6d0869c8b2f5/src/cryptography/hazmat/primitives/serialization/ssh.py#L648-L690) for an example where they are already decoding certs) with the exceptions of generating a _nonce_ which can be done with `secrets` and generating a _serial_ which `ssh-keygen` takes as user input anyway. It's just a matter of structuring certificate objects based on type as per the [spec](https://cvsweb.openbsd.org/src/usr.bin/ssh/PROTOCOL.certkeys?annotate=HEAD).

I personally use Vault for ssh key signing so it's not a major priority for me so outside of the 2 people from the original issue on ansible/ansible and the one person here I don't see any urgency to go with a suboptimal hack.